website.py

```diff
@@ -9,15 +9,14 @@
 # cgi tracebacks
 cgitb.enable()
 
-form = cgi.FieldStorage()
-do_visualize = "visualize" in form
-if do_visualize:
-    print(f"Content-Type: image/svg+xml; charset=utf-8")
-else:
-    print(f"Content-Type: text/html; charset=utf-8")
-print()
-sys.stdout.flush()
```

Comment:
The downside of delaying this is that if we hit an exception, cgitb will emit HTML before headers have been printed. Probably needs a try-catch